client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionDialog/InteractionEventForm/InteractionSection/RepetitiveEventForm/RepetitiveEventForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { isSameDay} from 'date-fns';
+import {Accordion, AccordionDetails, AccordionSummary, Button, Checkbox, FormControlLabel} from '@material-ui/core';
+import {ArrowDropDown, ArrowDropUp, ExpandMore} from '@material-ui/icons';
+import InteractionDetailsFields from '../InteractionDetailsFields/InteractionDetailsFields';
+import DetailsFieldsTitle from '../InteractionDetailsFields/DetailsFieldsTitle';
+import useStyles from './RepetitiveEventFormStyles';
+
+const MAX_DAYS_TO_DISPLAY = 7;
+const EDGE_DAYS_TO_DISPLAY = (MAX_DAYS_TO_DISPLAY - 1) / 2;
+
+const firstPossibleIndex = 0;
+const RepetitiveEventForm = ({datesToOffer, selectedDateIndex}: Props) => {
+    const [selectedDates, setSelectedDates] = React.useState<Date[]>([]);
+    const [firstDayToDisplay, setFirstDayToDisplay] = React.useState<number>(0);
+    const [lastDayToDisplay, setLastDayToDisplay] = React.useState<number>(0);
+    const classes = useStyles();
+
+    const lastPossibleIndex = React.useMemo(() => datesToOffer.length - 1, [datesToOffer]);
+
+    const getNextFutureIndex = (prevLastIndex: number) => {
+        return Math.min(lastPossibleIndex,prevLastIndex + EDGE_DAYS_TO_DISPLAY)
+    };
+
+    const getNextPastIndex = (prevFirstIndex: number) => {
+        return Math.max(firstPossibleIndex, prevFirstIndex - EDGE_DAYS_TO_DISPLAY)
+    };
+
+    React.useEffect(() => {
+        if (selectedDates.length === 0) {
+            setSelectedDates([datesToOffer[selectedDateIndex]]);
+            setFirstDayToDisplay(getNextPastIndex(selectedDateIndex));
+            setLastDayToDisplay(getNextFutureIndex(selectedDateIndex))
+        }
+    }, [selectedDateIndex]);
+
+    const loadNextDays = () => {
+        setLastDayToDisplay(getNextFutureIndex)
+    };
+
+    const loadPreviousDays = () => {
+        setFirstDayToDisplay(getNextPastIndex)
+    };
+
+    const onDateCheckClick = (date: Date, isChecked: boolean) => {
+        isChecked
+            ? setSelectedDates(prevDates => prevDates.concat(date))
+            : setSelectedDates(prevDates => prevDates.filter(selectedDay => !isSameDay(selectedDay, date)))
+    };
+
+    const isDateSelected = (date: Date) => selectedDates.some(selectedDay => isSameDay(selectedDay, date));
+
+    const DateCheckbox = ({day}: { day: Date }) => (
+        <Checkbox color='primary'
+                  checked={isDateSelected(day)}
+                  onChange={(event, checked) => onDateCheckClick(day, checked)}/>
+    );
+
+    const DateTitle = ({day}: { day: Date }) => (
+        <FormControlLabel
+            onClick={(event) => event.stopPropagation()}
+            onFocus={(event) => event.stopPropagation()}
+            control={<DateCheckbox day={day}/>}
+            label={<DetailsFieldsTitle date={day}/>}
+        />
+    );
```

Comment:
separate to component

Reply:
👍

Reply:
hmm, its already in a component did you mean move to different file?

Reply:
moved [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1236/commits/5365918a6fa6e3bef65c737949a82b6e8e62f1b8)

Reply:
yea tnx